documentation/workspace_manager.tex

```diff
@@ -203,19 +152,76 @@ \subsection{Program configuration}
 	\label{lst:pgm_conf}
 \end{listing}
 
+\section{Architecture overview}
+The architecture of the parser library is organized into the following components:
+
+\begin{description}
+	\item[Workspace manager API] The workspace manager provides API for handling various workspace management (e.g. add new workspace), LSP and DAP requests. It may hold multiple workspaces and calls 
+	file manager to handle changes in the workspace files.
+	\item[Workspace representation] The representation of workspace deals with the relations between its files (dependencies) upon parse request and propagates the parsing further into analyzer. It also retrieves data from the configuration files and it is used for resolving dependency searches by implementing parse library provider.
+	\item[Processor group representation] \sectionFiles{.45\linewidth}{parser\_library/src/workspaces/processor\_group.h} 
+	The representation of a processor group uses the API of libraries to search for their dependencies. Currently, we only support local libraries, which utilize the file manager for their file information retrieval.
+	\item[File manager] \sectionFiles{.45\linewidth}{parser\_library/src/workspaces/file\_manager.h}
+	The file manager is used by multiple components to handle file management and file searches. It also distinguishes and does conversions between regular files and processor files, which may be used for parsing.
+	\item[Analyzer] The analyzer accepts a file along with the information needed for dependency resolution, syntactically and semantically processes it and fills the context tables. The component is further explained in~\cref{chap:analyzer}.
+\end{description}
+
+An overview of the architecture is visualised in~\cref{ws_mngr_arch}.
+
+The technical details of each component are further explained in the following sections.
+
+\afterpage{
+\begin{landscape}
+\begin{figure}
+	\centering
+	\includegraphics[width=16cm]{img/ws_mngr_arch}
+	\caption{Architecture of workspace manager.}
+	\label{ws_mngr_arch}
+\end{figure}
+\end{landscape}
+}
+
+%******************** Workspace *************************
+\sectionSrc{Workspace representation}
+{parser\_library/src/workspaces/workspace.h}
+
+The representation of workspace is used by the workspace manager to handle various changes in the workspace. The workspace manager propagates LSP requests and notifications coming from the language server to the corresponding workspace and retrieves the results from it via the registered observers.
+
+The workspace component uses the file manager for the file searches, retrieves the values from the configuration files and creates processor groups and implements the parse library provider interface to be later used by the analyzer for the dependency resolution.
```

Comment:
myslim ze parse library provider je tu nedefinovany ci?

Reply:
ten sa definuje nizsie. mozem ho zadefinovat nejako vyssie aj ked neviem uplne ako. alebo vyhodit odtialto.

Reply:
asi vyhodit odtialto a radsej potom povedat ze workspace to riesi.. tu uz sme v poho definovali ze to riesi dependencie medzi files